From f1256eab089019910a7c99c8b2a31a1b1d2acdc4 Mon Sep 17 00:00:00 2001
From: Sebastian Reichel <sre@kernel.org>
Date: Wed, 2 Mar 2016 17:06:01 +0100
Subject: [PATCH 38/53] drm: omapdrm: crtc: update plane fifos on lcd config
 change

Due to a hardware bug, FIFOs thresholds must be
configured very carefully for manually updated
displays.

Signed-off-by: Sebastian Reichel <sre@kernel.org>
---
 drivers/gpu/drm/omapdrm/omap_crtc.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/omapdrm/omap_crtc.c b/drivers/gpu/drm/omapdrm/omap_crtc.c
index 337a7a0678b4..25df89457c39 100644
--- a/drivers/gpu/drm/omapdrm/omap_crtc.c
+++ b/drivers/gpu/drm/omapdrm/omap_crtc.c
@@ -256,6 +256,10 @@ static void omap_crtc_dss_set_lcd_config(enum omap_channel channel,
	omap_crtc->manually_updated = dss_lcd_mgr_config_get_stallmode(config);

	dispc_mgr_set_lcd_config(omap_crtc->channel, config);
+
+	drm_for_each_plane(plane, dev) {
+		omap_plane_update_fifo(plane);
+	}
 }

 static int omap_crtc_dss_register_framedone(
--
2.11.0
